Bottom: 4db764375aa98346b7793fbd8e07b152ddf66cd1
Top:    1f4f528788cab5d644bd344dd2456e0a2b3bd963
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-09-06 11:55:23 +0800

Refresh of vhost-prefetch-vring

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index c437a2c..acca816 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -460,7 +460,6 @@ static void handle_tx(struct vhost_net *net)
 	struct socket *sock;
 	struct vhost_net_ubuf_ref *uninitialized_var(ubufs);
 	bool zcopy, zcopy_used;
-	struct vring_desc descs[VHOST_RX_BATCH];
 	int i, batched = VHOST_RX_BATCH;
 
 	mutex_lock(&vq->mutex);
@@ -495,7 +494,7 @@ static void handle_tx(struct vhost_net *net)
 		if (unlikely(vhost_exceeds_maxpend(net)))
 			break;
 
-		avails = vhost_prefetch_desc_indices(vq, heads, descs,
+		avails = vhost_prefetch_desc_indices(vq, heads, vq->descs,
 						     batched, &cont);
 		/* On error, stop handling until the next kick. */
 		if (unlikely(avails < 0))
@@ -513,7 +512,7 @@ static void handle_tx(struct vhost_net *net)
 		}
 
 		for (i = 0; i < avails; i++) {
-			struct vring_desc *d = cont ? &descs[i] : NULL;
+			struct vring_desc *d = cont ? &vq->descs[i] : NULL;
 
 			head = __vhost_get_vq_desc(vq, vq->iov,
 						   ARRAY_SIZE(vq->iov),
diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index 9ea0b55..ca06fe2 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -149,6 +149,7 @@ struct vhost_virtqueue {
 	bool user_be;
 #endif
 	u32 busyloop_timeout;
+	struct vring_desc descs[64];
 };
 
 struct vhost_msg_node {
